PURL-SPECIFICATION.rst

```diff
@@ -482,22 +488,22 @@ pairs some of which may not be normalized:
 - **qualifiers**: the ``qualifiers`` corresponding to this ``purl`` as an object of
   {key: value} qualifier pairs.
 - **subpath**: the ``subpath`` corresponding to this ``purl``.
-- **is_invalid**: a boolean flag set to true if the test should report an
+- **is_invalid**: a boolean flag set to true if the test SHOULD report an
   error
 
 To test ``purl`` parsing and building, a tool can use this test suite and for
 every listed test object, run these tests:
 
 - parsing the test canonical ``purl`` then re-building a ``purl`` from these parsed
-  components should return the test canonical ``purl``
+  components SHOULD return the test canonical ``purl``
 
-- parsing the test ``purl`` should return the components parsed from the test
+- parsing the test ``purl`` SHOULD return the components parsed from the test
   canonical ``purl``
 
 - parsing the test ``purl`` then re-building a ``purl`` from these parsed components
-  should return the test canonical ``purl``
+  SHOULD return the test canonical ``purl``
 
-- building a ``purl`` from the test components should return the test canonical ``purl``
+- building a ``purl`` from the test components SHOULD return the test canonical ``purl``
```

Comment:
All of these are qualified MUSTs. The implementation MUST return the value specified by the test if the implementation supports the package type used by that test. The test suite could use some work to separate the behaviors that any PURL implementation must satisfy from the behaviors that PURL implementations supporting certain package types must satisfy.